app/Http/Controllers/SubmissionsController.php

```diff
@@ -2,32 +2,31 @@
 
 namespace App\Http\Controllers;
 
+use App\Conference;
+use App\Submission;
 use Illuminate\Http\Request;
-use App\Commands\CreateSubmission;
-use App\Commands\DestroySubmission;
 
 class SubmissionsController extends Controller
 {
     public function store(Request $request)
     {
-        $talk = auth()->user()->talks()->findOrFail($request->input('talkId'));
-
-        $this->dispatch(new CreateSubmission(
-            $request->input('conferenceId'),
-            $talk->id
-        ));
-
-        return response()->json(['status' => 'success', 'message' => 'Talk Submitted']);
+        $talk = auth()->user()->talks()->findOrFail($request->input('talkId'))->current();
+        $conference = Conference::findOrFail($request->input('conferenceId'));
+
+        $submission = Submission::create([
+            'conference_id' => $conference->id,
+            'talk_revision_id' => $talk->id
+        ]);
+        return response()->json([
+            'status' => 'success',
+            'message' => 'Talk Submitted',
+            'submissionId' => $submission->id
+        ]);
     }
 
-    public function destroy(Request $request)
+    public function destroy($id)
     {
-        $talk = auth()->user()->talks()->findOrFail($request->input('talkId'));
-
-        $this->dispatch(new DestroySubmission(
-            $request->input('conferenceId'),
-            $talk->id
-        ));
+        Submission::destroy($id);
```

Comment:
The original (using `auth()->user()->talks()`) guards that this user has access to this submission to delete it. This is a way we handled ACL before we had ACL actually coded in. :)